esp-wifi/src/ble/controller/mod.rs

```diff
@@ -3,22 +3,26 @@ use embedded_io::{Error, ErrorType, Read, Write};
 use super::{read_hci, read_next, send_hci};
 use crate::{
     hal::peripheral::{Peripheral, PeripheralRef},
-    EspWifiInitialization,
+    EspWifiController,
 };
 
 /// A blocking HCI connector
 pub struct BleConnector<'d> {
     _device: PeripheralRef<'d, crate::hal::peripherals::BT>,
 }
 
+impl<'d> Drop for BleConnector<'d> {
+    fn drop(&mut self) {
+        crate::ble::ble_deinit();
```

Comment:
Just a passing observation, is this really infallible?

Reply:
For BTDM `btdm_controller_deinit` doesn't return anything (and is the only function called)

For NPL most functions involved return results but the current implementation panics if something goes wrong